### PR TITLE
feat: scroll etherscan engine api

### PR DESCRIPTION
**Pull request by greged93**
Fixes the debug client implementation to work with all versions of the execution payload and adds a correct implementation for the `PayloadValidator` for Scroll, ensuring a correct conversion from payload to sealed block.

### CODE REVIEW

File: .github/assets/check_wasm.sh [hunk 1]
```diff
@@ -74,6 +74,7 @@ exclude_crates=(
   reth-scroll-cli # tokio
   reth-scroll-node # tokio
   reth-scroll # tokio
+  reth-scroll-engine # CI fails on tokio but compiles locally
```
Comment (greged93) [inline .github/assets/check_wasm.sh:77]: this is very puzzling: keeps failing in CI with `compile_error!("Only features sync,macros,io-util,rt,time are supported on wasm.");`, while it compiles fine on a linux server with the wasm target, using the same toolchain.

File: crates/consensus/debug-client/src/providers/etherscan.rs [hunk 2]
```diff
@@ -32,13 +32,17 @@ impl EtherscanBlockProvider {
     /// `BlockNumberOrTag::Earliest`, `BlockNumberOrTag::Pending`, `BlockNumberOrTag::Number(u64)`
     /// are supported.
     pub async fn load_block(&self, block_number_or_tag: BlockNumberOrTag) -> eyre::Result<Block> {
+        let tag = match block_number_or_tag {
+            BlockNumberOrTag::Number(num) => format!("{num:#02x}"),
+            tag => tag.to_string(),
+        };
```
Comment (frisitano) [inline crates/consensus/debug-client/src/providers/etherscan.rs:38]: Let's open an issue in upstream to see if we should change the `fmt::Display` implementation for this.

**Review by frisitano — APPROVED** [review]
Looks good! LGTM.

File: crates/consensus/debug-client/src/providers/etherscan.rs (hunk 2 above)
Comment (greged93) [reply crates/consensus/debug-client/src/providers/etherscan.rs:38]: https://github.com/alloy-rs/alloy/issues/1852

**Pull request merged**